scalableminds/webknossos
# Skip typing data chunk arrays if no rechunking is needed

In case the MultiArray chunks on disk are the same shape as the requested buckets, no rechunking is needed. There was already a performance shortcut for this to avoid the partial copy itself, but now we also skip typing the array at all (and converting it back to bytes).

### todo
 - [x] memory leak? (cache weight bug? or reference leaked?) → turned out the metadata shape was wrong, leading to the calculated chunk cache weight being 0, leading to the cache to grow without limits

### URL of deployed dev instance (used for testing):
- https://shortcutskiptyping.webknossos.xyz

### Steps to test:
- Load zarr dataset with nice 32³ buckets, should still look the same and be faster
- load any other zarr dataset, should still look the same

------
